libsolidity/analysis/ReferencesResolver.h

```diff
@@ -90,6 +94,7 @@ class ReferencesResolver: private ASTConstVisitor
 	std::vector<ParameterList const*> m_returnParameters;
 	bool const m_resolveInsideCode;
```

Comment:
What exactly is this setting?

Reply:
Reference resolution happens in two stages: In the first, all declarations at contract level are registered, but nothing inside code. In the second, code is also examined. This is necessary since at contract level, any declaration can appear anywhere, and for some parts - I don't remember anymore what exactly - it is important at this stage whether an identifier is a type name or e.g. a function.